src/core/http/request_template.rs

```diff
@@ -53,7 +54,7 @@ impl RequestTemplate {
 
         let qp_string = base_qp
             .chain(extra_qp)
-            .map(|(k, v)| format!("{}={}", k, v))
+            .map(|(k, v)| ctx.encode(&k, &v))
```

Comment:
Don't call `encode` on query param, instead implement a special query param encoder that handles mustache templates by implementing `mustache::Eval`